admin/module/catalog-admin-module-sources/src/main/java/org/codice/ddf/catalog/admin/plugin/AdminSourcePollerServiceBean.java

```diff
@@ -0,0 +1,280 @@
+                        } else {
+                            source.put(MAP_ENTRY_NAME, config.getPid());
+                        }
+
+                        Dictionary<String, Object> properties = config.getProperties();
+                        Map<String, Object> plist = new HashMap<>();
+                        for (String key : Collections.list(properties.keys())) {
+                            plist.put(key, properties.get(key));
+                        }
+                        source.put(MAP_ENTRY_PROPERTIES, plist);
+
+                        configurations.add(source);
+                    }
+                    metatype.put(MAP_ENTRY_CONFIGURATIONS, configurations);
+                }
+            } catch (Exception e) {
+                LOGGER.warn("Error getting source info: {}", e.getMessage());
+            }
+        }
+
+        Collections.sort(metatypes, new Comparator<Map<String, Object>>() {
+            @Override
+            public int compare(Map<String, Object> o1, Map<String, Object> o2) {
+                return ((String) o1.get("id")).compareToIgnoreCase((String) o2.get("id"));
+            }
+        });
+        return metatypes;
+    }
+
+    protected OsgiHelper getHelper() {
+        return new OsgiHelper();
+    }
+
+    public class OsgiHelper {
+        protected ConfigurationAdmin configurationAdmin;
+
+        protected BundleContext getBundleContext() {
+            Bundle bundle = FrameworkUtil.getBundle(AdminSourcePollerServiceBean.class);
+            if (bundle != null) {
+                return bundle.getBundleContext();
+            }
+            return null;
+        }
+
+        protected List<ServiceReference<? extends Source>> getServiceReferences()
+                throws org.osgi.framework.InvalidSyntaxException {
+            List<ServiceReference<? extends Source>> refs = new ArrayList<>();
+
+            refs.addAll(
+                    helper.getBundleContext().getServiceReferences(FederatedSource.class, null));
+            refs.addAll(
+                    helper.getBundleContext().getServiceReferences(ConnectedSource.class, null));
+            return refs;
+        }
+
+        protected List<Map<String, Object>> getMetatypes() {
+            ConfigurationAdminExt configAdminExt = new ConfigurationAdminExt(configurationAdmin);
+            return configAdminExt.addMetaTypeNamesToMap(configAdminExt.getFactoryPidObjectClasses(),
+                    "(|(service.factoryPid=*source)(service.factoryPid=*Source)(service.factoryPid=*service)(service.factoryPid=*Service))",
+                    "service.factoryPid");
+        }
+
+        protected Map getNameMap() {
+            ConfigurationAdminExt configAdminExt = new ConfigurationAdminExt(configurationAdmin);
+            return configAdminExt.getFactoryPidObjectClasses();
+        }
+
+        protected Configuration[] getConfigurations(Map metatype)
```

Comment:
You can also hide OSGi details in this method by returning a List<Map<String, Object>> instead of Configuration[].